src/middlewared/middlewared/main.py

```diff
@@ -734,29 +734,22 @@ def __setup_periodic_tasks(self):
                         delay,
                         functools.partial(
                             self.__call_periodic_task,
-                            method, service_name, task_name, method._periodic.interval
+                            method, service_name, service_obj, task_name, method._periodic.interval
                         )
                     )
 
-    def __call_periodic_task(self, method, service_name, task_name, interval):
-        self.__loop.create_task(self.__periodic_task_wrapper(method, service_name, task_name, interval))
+    def __call_periodic_task(self, method, service_name, service_obj, task_name, interval):
+        self.__loop.create_task(self.__periodic_task_wrapper(method, service_name, service_obj, task_name, interval))
 
-    async def __periodic_task_wrapper(self, method, service_name, task_name, interval):
+    async def __periodic_task_wrapper(self, method, service_name, service_obj, task_name, interval):
         self.logger.trace("Calling periodic task %s::%s", service_name, task_name)
-
-        try:
-            if asyncio.iscoroutinefunction(method):
-                await method()
-            else:
-                await self.run_in_thread(method)
-        except Exception:
-            self.logger.warning("Exception while calling periodic task", exc_info=True)
+        await self._call(task_name, service_obj, method)
```

Comment:
@650elx  Dont we still need a try/except block here? Or a failed task will not schedule the next run?

Reply:
You're right - it's fixed now.